source/MaterialXRuntime/RtFileIo.cpp

```
@@ -563,40 +617,58 @@ RtApiType RtFileIo::getApiType() const
     return RtApiType::CORE_IO;
 }
 
-void RtFileIo::read(const FilePath& documentPath, const FileSearchPath& searchPaths, RtFileIo::ReadFilter filter)
+
+void RtFileIo::read(const FilePath& documentPath, const FileSearchPath& searchPaths, const RtReadOptions* readOptions)
 {
     try
     {
         DocumentPtr document = createDocument();
-        XmlReadOptions readOptions;
-        readOptions.skipConflictingElements = true;
-        readFromXmlFile(document, documentPath, searchPaths, &readOptions);
+        XmlReadOptions xmlReadOptions;
+        xmlReadOptions.skipConflictingElements = true;
+        if (readOptions)
+        {
+            xmlReadOptions.skipConflictingElements = readOptions->skipConflictingElements;
+        }
+        readFromXmlFile(document, documentPath, searchPaths, &xmlReadOptions);
 
         PrvStage* stage = data()->asA<PrvStage>();
-        readDocument(document, stage, filter);
+        readDocument(document, stage, readOptions);
     }
-    catch(Exception&)
+    catch (Exception&)
     {
         return;
```

Not part of this review but should we silently hide all exceptions here?